executor/infoschema_reader.go

```diff
@@ -2616,54 +2614,73 @@ type tiflashInstanceInfo struct {
 }
 
 func (e *TiFlashSystemTableRetriever) initialize(sctx sessionctx.Context, tiflashInstances set.StringSet) error {
-	store := sctx.GetStore()
-	if etcd, ok := store.(kv.EtcdBackend); ok {
-		var addrs []string
-		var err error
-		if addrs, err = etcd.EtcdAddrs(); err != nil {
+	storeInfo, err := infoschema.GetStoreServerInfo(sctx)
+	if err != nil {
+		return err
+	}
+
+	for _, info := range storeInfo {
+		if info.ServerType != kv.TiFlash.Name() {
+			continue
+		}
+		info.ResolveLoopBackAddr()
+		if len(tiflashInstances) > 0 && !tiflashInstances.Exist(info.Address) {
+			continue
```

Comment:
append a warning?

Reply:
`tiflashInstances` is just the where clause in the select list. 
For example, if the sql is `select * from tiflash_tables`, then `tiflashInstances` is empty and it means we need to select the info for all tiflash nodes in the cluster.
If the sql is `select * from tiflash_tables where tiflash_instance=127.0.0.1:9000`, then `tiflashInstances` contains the element `127.0.0.1:9000` and we need filter out other tiflash nodes.